### ISSUE TITLE
Query fails with newMemoryReservation is negative

### ISSUE THREAD

**Issue by nezihyigitbasi** [prestodb/presto#8341]
```
java.lang.IllegalArgumentException: newMemoryReservation is negative
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)
        at com.facebook.presto.operator.OperatorContext.setMemoryReservation(OperatorContext.java:306)
        at com.facebook.presto.operator.AggregationOperator.addInput(AggregationOperator.java:168)
        at com.facebook.presto.operator.Driver.processInternal(Driver.java:309)
        at com.facebook.presto.operator.Driver.lambda$processFor$6(Driver.java:234)
        at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:541)
        at com.facebook.presto.operator.Driver.processFor(Driver.java:229)
        at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:623)
        at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
        at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:463)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by nezihyigitbasi** [comment]
I don't have a minimal example to repro this unfortunately.

**Comment by nezihyigitbasi** [comment]
`SELECT cardinality(array_distinct(flatten(array_agg(x)))) FROM T WHERE ds = '<ds>';`

**Comment by nezihyigitbasi** [comment]
This bug was reported on v 0.179. It turns out that there is an integer overflow [here](https://github.com/prestodb/presto/blob/0.179/presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java#L104). The good thing is this will be fixed in 0.180 as a result of #8124.
